docs/DafnyRef/UserGuide.md

```diff
@@ -51,129 +96,315 @@ verification failure in a file that is being checked.
+with an unsupported suffix, provoking an error message..
+- Files containing dafny code must have a `.dfy` suffix.
+- There must be at least one `.dfy` file.
+- The command-line may contain other kinds of files appropriate to
+the language that the Dafny files are being compiled to. The kind of file is determined by its suffix.
+- Escape characters are determined by the shell executing the command-line.
+- The option `--` means that all subsequent command-line arguments are not options to the dafny tool; they are either files or arguments to the `dafny run` command.
+- If an option is repeated (e.g., with a different argument), then the later instance on the command-line supersedes the earlier instance, with just a few options accumulating arguments.
+- If an option takes an argument, the option name is followed by a `:` or `=` or whitespace and then by the argument value; if the argument itself contains white space, the argument must be enclosed in quotes.
+- Boolean options can take the values `true` and `false` (or any case-insensitive version of those words). For example, the value of `--no-verify` is by default `false` (that is, do verification). It can be set to true (no verification) using `--no-verify`, `--no-verify:true`, `--no-verify=true`, `--noverify true`; it can be set false (do verification) using `--no-verify:false` or `--no-verify=false` or `--no-verify false`.
+- There is a potential ambiguity when the form `--option value` is used if the value is optional (such as for boolean values). In such a case an argument afer an option (that does not have an argument given with `:` or `=`) is interpreted as the value if it is indeed a valid value for that option. However, better style advises always using a ':' or '=' to set option values.
+No valid option values in dafny look like filenames or begin with `--`.
+
+
+Options may change over time; `dafny <command> --help` lists those options for a given \<command\> that are implemented in your version of dafny.
+
+#### 25.5.1.1. Options that are not associated with a command
+
+A few options are not part of a command. In these cases any single-hyphen spelling also pewrmits a spelling beginning with '/'.
+- `dafny --help` or `dafny -h` lists all the available commands
+- `dafny -?` or `dafny -help` list all legacy options
+- `dafny --version` (or `-version`) prints out the numbe of the version this build of dafny implements
+- `dafny -attrHelp` prints out help information about attributes
+
+
+#### 25.5.1.2. `dafny resolve` {#sec-dafny-resolve}
+
+The `dafny resolve` command checks the command-line and then parses and typechecks the given files and any included files.
+
+The options relevant to this command are
+- those relevant to the command-line itself
+   - `--` --- signals that any command-line arguments after the appearance of `--` are interpreted as files and not as options (but see [`dafny run`](#sec-dafny-run) for a variation)
```

Comment:
I would mention `--` at most once since this is part of POSIX so it already doesn't really belong in a Dafny manual.

Reply:
OK